### ISSUE TITLE
Port twisted.python.systemd to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7419]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7419|
|Type|enhancement|
|Created|2014-05-25 11:17:39Z|
|Branch|https://github.com/twisted/twisted/tree/port-systemd-7419|

Rather than disable it, maybe we should port it.

<details><summary>Searchable metadata</summary>

```
trac-id__7419 7419
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_port_systemd_7419 branches/port-systemd-7419
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1401016659000000 1401016659000000
changetime__1445217216921098 1445217216921098
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [42743]) Branching to port-systemd-7419.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl removed owner|
|-|-|

This would fix [#7414](https://github.com/twisted/twisted/issues/7414) by porting it rather than removing it for 14.1.

**Comment by twisted-trac** [comment]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|Alex set owner to @hawkowl|
|-|-|

There is a twisted checker failure which is relevant, and a failure on the py3k builder. That failure doesn't /look/ relevant, but please take a look and confirm my suspicions. Besides those this LGTM, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

The failure is due to the service_identity stuff.

I'll fix up the twistedchecker problem and merge it.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

(In [42747]) Merge port-systemd-7419: Port twisted.python.systemd to Python 3.

Author: hawkowl
Reviewer: Alex
Fixes: [#7419](https://github.com/twisted/twisted/issues/7419)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_7"></a>|@hawkowl commented|
|-|-|

Ticket retargeted after milestone deleted
